pkg/controllers/node/drift.go

```diff
@@ -73,3 +75,24 @@ func (d *Drift) Reconcile(ctx context.Context, provisioner *v1alpha5.Provisioner
 	// Requeue after 5 minutes for the cache TTL
 	return reconcile.Result{RequeueAfter: 5 * time.Minute}, nil
 }
+
+// isProvisionerDrifted will increase in scope to include more fields of the provisioner
+func (d *Drift) isProvisionerDrifted(ctx context.Context, provisioner *v1alpha5.Provisioner, machine *v1alpha5.Machine) (bool, error) {
+	cloudProviderDrifted, err := d.cloudProvider.IsMachineDrifted(ctx, machine)
+	if err != nil {
+		return false, fmt.Errorf("getting drift for node, %w", err)
+	}
+	nodeRequirementsDrifted := isNodeRequirementsDrifted(machine, provisioner)
+
+	return cloudProviderDrifted || nodeRequirementsDrifted, nil
+}
+
+func isNodeRequirementsDrifted(machine *v1alpha5.Machine, provisioner *v1alpha5.Provisioner) bool {
+	machineRequirements := scheduling.NewNodeSelectorRequirements(machine.Spec.Requirements...)
+	provisionerRequirements := scheduling.NewNodeSelectorRequirements(provisioner.Spec.Requirements...)
+
+	compatibilityProvisionerToMachine := provisionerRequirements.Compatible(machineRequirements)
+	compatibilityMachineToProvisioner := machineRequirements.Compatible(provisionerRequirements)
+
+	return compatibilityProvisionerToMachine != nil || compatibilityMachineToProvisioner != nil
```

Comment:
Just curious: What's the reason we are going both ways with this compatibility check?